### PR TITLE
Update standard template to have pay processor specific values

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-806

### Change description ###

- This PR includes

1. Updating standard spring boot template with pay processor specific values.

2. Jenkins_CNP and Jenkins_parameterized

3. Helm chart updates

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@Library("Infrastructure")
+
+import uk.gov.hmcts.contino.GradleBuilder
+import uk.gov.hmcts.contino.Kubectl
+import uk.gov.hmcts.contino.DockerImage
+import uk.gov.hmcts.pipeline.TeamConfig
+
+def type = "java"
+def product = "bulk-scan"
+def component = "pay-processor"
```
Comment (doncem) [inline Jenkinsfile_CNP]: because `payment-processor` is too long i presume, right?

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!groovy
+@Library("Infrastructure")
+
+import uk.gov.hmcts.contino.GradleBuilder
+import uk.gov.hmcts.contino.Kubectl
+import uk.gov.hmcts.contino.DockerImage
+import uk.gov.hmcts.pipeline.TeamConfig
```
Comment (doncem) [inline Jenkinsfile_CNP]: only `GradleBuilder` is needed. rest can be removed 😉

File: Jenkinsfile_parameterized [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!groovy
+@Library(value="Infrastructure", changelog=false) _
```
Comment (doncem) [inline Jenkinsfile_parameterized]: no need for `_` at the end

File: Jenkinsfile_parameterized [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+properties([
+  parameters([
+    string(name: 'PRODUCT_NAME', defaultValue: 'bulk-scan', description: ''),
+    string(name: 'APP', defaultValue: 'pay-processor', description: ''),
+    string(name: 'TYPE', defaultValue: 'java', description: ''),
+    string(name: 'ENVIRONMENT', defaultValue: 'sandbox', description: 'Environment where code should be built and deployed'),
+    choice(name: 'SUBSCRIPTION', choices: 'sandbox\nnonprod\nprod', description: 'Azure subscriptions available to build in')
+  ]),
+  [$class: 'GithubProjectProperty', projectUrlStr: 'github.com/hmcts/bulk-scan-ccd-event-handler-sample-app'],
```
Comment (doncem) [inline Jenkinsfile_parameterized]: incorrect project url

File: Jenkinsfile_CNP (hunk 1 above)
Comment (nitinprabhu) [reply Jenkinsfile_CNP]: yes that's correct

File: Jenkinsfile_parameterized (hunk 4 above)
Comment (nitinprabhu) [reply Jenkinsfile_parameterized]: fixed now

File: charts/bulk-scan-payment-processor/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+appVersion: "1.0"
+description: A helm chart for bulk scan pay processor app
+name: bulk-scan-payment-processor
+home: https://github.com/hmcts/bulk-scan-payment-processor
+version: 0.0.3
```
Comment (doncem) [inline charts/bulk-scan-payment-processor/Chart.yaml]: aye, perhaps version `0.0.1` as of now. if not `0.0.0` 😂

File: infrastructure/main.tf [hunk 6]
```diff
@@ -1 +1,26 @@
...
\ No newline at end of file
+provider "azurerm" {}
+
+locals {
+  ase_name            = "${data.terraform_remote_state.core_apps_compute.ase_name[0]}"
+  is_preview          = "${(var.env == "preview" || var.env == "spreview")}"
+  local_env           = "${local.is_preview ? "aat" : var.env}"
+  s2s_rg              = "rpe-service-auth-provider-${local.local_env}"
+  s2s_url             = "http://${local.s2s_rg}.service.core-compute-${local.local_env}.internal"
+  ccd_api_url         = "http://ccd-data-store-api-${local.local_env}.service.core-compute-${local.local_env}.internal"
```
Comment (doncem) [inline infrastructure/main.tf]: so sad this service needs the dependency on this one 😢

File: src/main/resources/application.yaml [hunk 7]
```diff
@@ -1,33 +1,11 @@
 server:
-  port: 4550
+  port: 8583
 
 management:
   endpoints:
```
Comment (doncem) [inline src/main/resources/application.yaml]: for successful PR i believe we need the following under `endpoints`:

```yaml
health:
  show-details: "always"
```

File: charts/bulk-scan-payment-processor/Chart.yaml (hunk 5 above)
Comment (nitinprabhu) [reply charts/bulk-scan-payment-processor/Chart.yaml]: oops that's from spring boot template :P

File: infrastructure/main.tf (hunk 6 above)
Comment (nitinprabhu) [reply infrastructure/main.tf]: will remove this for now as we don't know how we are going to implement it

**Pull request merged**